server/src/main/java/org/elasticsearch/action/bulk/TransportBulkAction.java

```diff
@@ -523,26 +524,16 @@ protected void doRun() {
                         throw new IllegalArgumentException("only write ops with an op_type of create are allowed in data streams");
                     }
 
-                    IndexRouting indexRouting = concreteIndices.routing(concreteIndex);
-
-                    switch (docWriteRequest.opType()) {
-                        case CREATE:
-                        case INDEX:
-                            prohibitAppendWritesInBackingIndices(docWriteRequest, metadata);
-                            prohibitCustomRoutingOnDataStream(docWriteRequest, metadata);
-                            IndexRequest indexRequest = (IndexRequest) docWriteRequest;
-                            indexRequest.resolveRouting(metadata);
-                            indexRequest.process();
-                            break;
-                        case UPDATE:
-                            docWriteRequest.routing(metadata.resolveWriteIndexRouting(docWriteRequest.routing(), docWriteRequest.index()));
-                            break;
-                        case DELETE:
-                            docWriteRequest.routing(metadata.resolveWriteIndexRouting(docWriteRequest.routing(), docWriteRequest.index()));
-                            break;
-                        default:
-                            throw new AssertionError("request type not supported: [" + docWriteRequest.opType() + "]");
+                    if (docWriteRequest.opType() == OpType.CREATE || docWriteRequest.opType() == OpType.INDEX) {
+                        prohibitAppendWritesInBackingIndices(docWriteRequest, metadata);
+                        prohibitCustomRoutingOnDataStream(docWriteRequest, metadata);
+                    }
+                    docWriteRequest.routing(metadata.resolveWriteIndexRouting(docWriteRequest.routing(), docWriteRequest.index()));
+                    if (docWriteRequest.opType() == OpType.CREATE || docWriteRequest.opType() == OpType.INDEX) {
+                        ((IndexRequest) docWriteRequest).process();
```

Comment:
Why not add `DocWriteRequest.process()` too to avoid the `if` here?

Reply:
I was looking at that. Flipping it over and over like it was a serving bowl I didn't need at a yard sale. I'll put it back that way and see if you like.

Reply:
Ah ha! The thing that was bothering me last time was that I tried to merge the alias routing setting *into* `DocWriteRequest.process` but that wasn't working right because we need to call `DocWriteRequest.process` on the data nodes after update requests have been translated. I didn't think to myself "what if I didn't try to merge those things?"